weather-gov/weather.gov
# Make forecast details buttons WCAG tabs (a11y)

## What does this PR do? 🛠️
As discussed in #1944, the toggle buttons for switching between charts and the table for forecast details are not accessible. This PR does several things (noted below), but mainly changes those buttons to be [WCAG compliant tabs](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/).

## What does the reviewer need to know? 🤔
### Generic Tabs component, and proposal for cleanup ###
Right now, we have **three** separate components that each implement the tabs functionality. This is a lot of repetition, especially when it comes to binding the keyboard navigation events.
  
For this work, we have created a _generic_ Tab component (`wx-tabs`) that we hope the other tab components can be refactored from. We've even made a [ticket for that work here](https://github.com/weather-gov/weather.gov/issues/2152).

### Tests ###
We have added some JSDom based component tests. We need to make sure these are running properly as part of the code-standards CI.
  
### Other issues ###
We had to modify some of the markup. There was one place in which all of the forecast details tables had the same id across the document -- huge NO-NO!

## How to test?
Pull down this branch and then use VoiceOver to ensure that the Table/Charts button group is being interpreted as a tablist. Test the keyboard navigation events that are specified in the [WCAG recommendation](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/).

## Screenshots (if appropriate): 📸
Nope -- it looks exactly the same!
